tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/ControlBuilder.scala

```diff
@@ -0,0 +1,48 @@
+package at.forsyte.apalache.tla.typecomp.subbuilder
+
+import at.forsyte.apalache.tla.typecomp.BuilderUtil._
+import at.forsyte.apalache.tla.typecomp._
+import at.forsyte.apalache.tla.typecomp.unsafe.UnsafeControlBuilder
+import scalaz._
+
+/**
+ * Type-safe builder for TlaSetOper expressions.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+trait ControlBuilder extends UnsafeControlBuilder {
+
+  /** IF p THEN A ELSE B */
+  def ite(p: TBuilderInstruction, A: TBuilderInstruction, B: TBuilderInstruction): TBuilderInstruction =
+    tertiaryFromUnsafe(p, A, B)(_ite)
+
+  /** CASE p1 -> e1 [] p2 -> e2 [] ... [] pn -> en */
+  def caseSplit(pairs: (TBuilderInstruction, TBuilderInstruction)*): TBuilderInstruction =
+    caseSplitMixed(pairs.flatMap { case (a, b) => Seq(a, b) }: _*)
+
+  /**
+   * Alternate call method, where paris are passed mixed
```

Comment:
Not sure "mixed" is very descriptive, can we say "interleaved"?
```suggestion
   * Alternate call method, where pairs are passed interleaved
```

Reply:
While that's more accurate, it's also more verbose, and we're trying to keep builder method names short (unless they should be used cautiously)

Reply:
Method name is fine with me, this was about the Scaladoc

Reply:
Oh, sure, we can change that

Reply:
Actually, can't you overload the method though?

Reply:
No, because erasure prevents you from overloading variadics

Reply:
Right, my bad!